extensions/data-workspace/src/dialogs/projectDashboard.ts

```diff
@@ -0,0 +1,124 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import { IProjectAction, IProjectActionGroup, IProjectProvider, WorkspaceTreeItem } from 'dataworkspace';
+import { IWorkspaceService } from '../common/interfaces';
+
+export class ProjectDashboard {
+
+	private dashboard: azdata.window.ModelViewDashboard | undefined;
+	private modelView: azdata.ModelView | undefined;
+	private projectProvider: IProjectProvider | undefined;
+	private overviewTab: azdata.DashboardTab | undefined;
+
+	constructor(private workspaceService: IWorkspaceService, private treeItem: WorkspaceTreeItem) {
+	}
+
+	public async showDashboard(): Promise<void> {
+		const project = this.treeItem.element.project;
+		this.projectProvider = await this.workspaceService.getProjectProvider(vscode.Uri.file(project.projectFilePath));
+		if (!this.projectProvider) {
+			throw new Error(constants.ProviderNotFoundForProjectTypeError(project.projectFilePath));
+		}
+
+		await this.createDashboard(project.projectFileName);
+		await this.dashboard!.open();
+	}
+
+	private async createDashboard(title: string): Promise<void> {
+		this.dashboard = azdata.window.createModelViewDashboard(title, 'ProjectDashboard', { alwaysShowTabs: false });
+		this.dashboard.registerTabs(async (modelView: azdata.ModelView) => {
+			this.modelView = modelView;
+
+			this.overviewTab = {
+				title: '',
+				id: 'overview-tab',
+				content: this.createContainer(),
+				toolbar: this.createToolbarContainer()
+			};
+			return [
+				this.overviewTab
+			];
+		});
+	}
+
+	private createToolbarContainer(): azdata.ToolbarContainer {
+		const projectActions: (IProjectAction | IProjectActionGroup)[] = this.projectProvider!.projectActions;
+
+		// Add actions as buttons
+		const buttons: azdata.ToolbarComponent[] = [];
+
+		projectActions.forEach(action => {
```

Comment:
Just a style thing, but this seems like an odd place for a functional-style loop since the lambda's more than just a line or three.  Any reason you didn't use `for-of`?

Reply:
Didn't use for-of since I needed a track of index (made after Alan's comment). I could have used for-in, but just to keep fetching of elements cleaner, using forEach for now.